examples/bri-3/src/bri/zkp/agents/proofStorage.agent.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { BlockchainService } from '../../../../blockchain/blockchain.service';
+
+import { Proof } from '../models/proof';
+
+@Injectable()
+export class ProofStorageAgent extends BlockchainService {
+  async storeProofInShieldContract(proof: Proof): Promise<void> {
+    await this.store(proof.payload);
+  }
+
+  async verifyProofInShieldContract(proofToVerify: string): Promise<boolean> {
```

Comment:
should proofToVerify be string or `Proof` here?

Reply:
it needs to be the public witness so public inputs + proof + verification key

Reply:
As @Therecanbeonlyone1969 has mentioned, the standards specify that for verification public inputs + proof + verification key are required. But in our implementation, the part of the proof that needs to be verified (i.e. content-addressable hash) is already stored onchain. Since we are not using zero-knowledge proofs atm, I have not added the verification key. I will rename "proofToVerify" --> "publicInputForProofVerification"

Reply:
> As @Therecanbeonlyone1969 has mentioned, the standards specify that for verification public inputs + proof + verification key are required. But in our implementation, the part of the proof that needs to be verified (i.e. content-addressable hash) is already stored onchain. Since we are not using zero-knowledge proofs atm, I have not added the verification key. I will rename "proofToVerify" --> "publicInputForProofVerification"

I would keep things consistent such that we can add more prover systems in the future without having to rename things. I would call it `publicWitnessForProofVerification`